### PR TITLE
Added ActionSpaceConfig

### LINKED ISSUE #26: [Proposal] Alternative action spaces
### Proposal 

Add a way to customize the action space.

### Motivation

Previous works on MiniWoB++ use different action spaces:

* Shi17: [World of Bits: An Open-Domain Platform for Web-Based Agents](http://proceedings.mlr.press/v70/shi17a/shi17a.pdf)
  - Action type: multinomial over {no-op, click, drag, scroll-up, scroll-down, key-press}
  - Cursor position: multinomial over [0, W) × [0, H) grid
  - Keyboard key: multinomial over keys + some allowed shortcuts
* Liu18: [Reinforcement Learning on Web Interfaces Using Workflow-Guided Exploration](https://arxiv.org/abs/1802.08802), Gur18: [Learning to Navigate the Web](https://arxiv.org/abs/1812.09195), and Jia19: [DOM-Q-NET: Grounded RL on Structured Language](https://arxiv.org/abs/1902.07257)
  - Action type: multinomial over {click, focus-and-type}
  - Element: multinomial over elements
  - Typed text: multinomial over field values
* Humphreys22: [A data-driven approach for learning to control computers](https://arxiv.org/abs/2202.08137)
  - Action type: multinomial over 11 {no-op, move, click, double-click, press, release, wheel-up, wheel-down, key-press, emit-text}
  - Cursor position: multinomial over 51 bins in each dimension
  - Keyboard key (for key-press): multinomial over keys + some allowed shortcuts
  - Task field (for emit-text): multinomial over field values

Many RL methods work best on discrete action spaces, but it trades off with generality (e.g., binned cursor position vs any cursor position).

### Pitch

Add ActionSpaceConfig that allows the user to customize:

* The list of action types to support
* Whether the cursor position should be binned
* The list of allowed keyboard keys
* Whether the emitted text is freeform or selected from a task field

### Alternatives

One alternative is to always include all action types in the space (including redundant ones like clicking elements + clicking coordinates). Downsides include:
* The RL agent can no longer define a simple multinomial distribution over the action types (a mask is needed).
* If we add new action types, the action indices will change.

### Checklist

- [x] I have checked that there is no similar [issue](https://github.com/Farama-Foundation/miniwob-plusplus/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by ppasupat**
# Description

Added ActionSpaceConfig for custom action spaces.

- Select a custom list of actions to include.
- For coordinates, added support for binned coordinates (divide the screen with a grid, then click the middle of a grid cell).

Fixes #26.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `pre-commit run --all-files` (see `CONTRIBUTING.md` instructions to set it up)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: miniwob/action.py [hunk 1]
```diff
@@ -1,156 +1,183 @@
...
+                    ActionTypes.CLICK_ELEMENT,
+                    ActionTypes.TYPE_TEXT,
+                    ActionTypes.FOCUS_ELEMENT_AND_TYPE_TEXT,
+                ]
+            )
+        else:
+            raise ValueError(f"Unknown preset name {name}")
+
+
+def get_action_space(config: ActionSpaceConfig) -> spaces.Space:
```
Comment (pseudo-rnd-thoughts) [inline miniwob/action.py:116]: Do we want `get_action_space` and `get_raw_coords` as `ActionSpaceConfig` methods

File: pyproject.toml [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 include = [
     "miniwob/**",
+    "tests/**",
```
Comment (pseudo-rnd-thoughts) [inline pyproject.toml:5]: I don't think we should be including `tests` in the release

**Review by pseudo-rnd-thoughts — APPROVED** [review]
Changes look good, I just have a question on if the functions should be methods or just functions

File: pyproject.toml (hunk 2 above)
Comment (pseudo-rnd-thoughts) [reply pyproject.toml:5]: Sorry I thought this line was for the files that are included in the release.
You are correct, this is the correct way of adding tests to pyright.
Though I should note that we normally don't care about having correct type hints for the tests

Comment (ppasupat) [reply pyproject.toml:5]: Thanks. It still helps with catching mistakes like missing method arguments, so I would prefer to keep it here for now.

File: miniwob/action.py (hunk 1 above)
Comment (ppasupat) [reply miniwob/action.py]: I have moved the methods to ActionSpaceConfig.

**Pull request merged**